### ISSUE TITLE
Case of Genesis Transaction 092111e882 ... c41cc

### ISSUE THREAD

**Issue by cypherpork** [ordinals/ord#2313]
Take a look at [this](https://blockstream.info/tx/092111e882a8025f3f05ab791982e8cc7fd7395afe849a5949fd56255b5c41cc?expand) transaction: according to Blockstream Explorer, it has 116 inputs and one output; Ord Explorer is [in agreement with Blockstream](https://ordinals.com/tx/092111e882a8025f3f05ab791982e8cc7fd7395afe849a5949fd56255b5c41cc). 

At the same time, the 116 inscriptions that one can find in the witness data are referred to by Ord Explorer as:

https://ordinals.com/inscription/092111e882a8025f3f05ab791982e8cc7fd7395afe849a5949fd56255b5c41cci0 - a non-cursed inscription 551694
Plus 115 cursed:
https://ordinals.com/inscription/092111e882a8025f3f05ab791982e8cc7fd7395afe849a5949fd56255b5c41cci1 - a cursed inscription -395
...
https://ordinals.com/inscription/092111e882a8025f3f05ab791982e8cc7fd7395afe849a5949fd56255b5c41cci115 - a cursed inscription -509

If the index number after "i" in the inscription URL is the number of the output of its genesis transaction, then the `i1` ... `i115` URLs contradict Blockstream and Ord Explorers showing this transaction having only one output.

**Comment by gmart7t2** [comment]
The number after the `i` is a count of inscription envelopes in that tx. The first inscription is `i0`. The 2nd is `i1`, and so on.

**Comment by cypherpork** [comment]
@gmart7t2, how you define the index number after the `i` makes a lot of sense – it points directly to where the content is stored in the witness.  Let me rephrase to make sure I understand this correctly. 

*The index number after `i` is the input number of the genesis transaction, where such input contains the witness with the inscribed content inside.* 

If there is only one input in the genesis, the index is `0`.  If it has `n` inputs, the index will be between `0` and `n-1`.

Please confirm that I formulated this correctly. 

If I did, I would close this issue and most likely open another one: I believe I saw several inscriptions that the Ord Explorer presents in a way that contradicts this new definition (to be confirmed).

**Comment by veryordinally** [comment]
@cypherpork Your rephrasing isn't aligned with the definition. The index following the i indicates the position of the inscription envelope within the entire transaction, irrespective of its specific input.

@raphjaph this is related to the inscription index stability issue we discussed this morning.

**Comment by cypherpork** [comment]
@veryordinally - something like this?

<table class="tg">
<thead>
  <tr>
    <th class="tg-3pgb">Input Number</th>
    <th class="tg-3pgb">Envelopes</th>
    <th class="tg-3pgb">Index</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-ftz9" rowspan="2">0</td>
    <td class="tg-3guk"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-4ntm">i0</td>
  </tr>
  <tr>
    <td class="tg-3guk"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-4ntm">i1</td>
  </tr>
  <tr>
    <td class="tg-fvke">1</td>
    <td class="tg-ll5l"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-e6uf">i2</td>
  </tr>
  <tr>
    <td class="tg-qk7o" rowspan="3">2</td>
    <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-edzw">i3</td>
  </tr>
  <tr>
    <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-edzw">i4</td>
  </tr>
  <tr>
    <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-edzw">i5</td>
  </tr>
  <tr>
    <td class="tg-fvke">3</td>
    <td class="tg-e6uf">none</td>
    <td class="tg-e6uf"></td>
  </tr>
  <tr>
    <td class="tg-mbap">4</td>
    <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
    <td class="tg-edzw">i6</td>
  </tr>
</tbody>
</table>

**Comment by veryordinally** [comment]
> @veryordinally - something like this?
> 
> 
> 
> <table class="tg">
> 
> <thead>
> 
>   <tr>
> 
>     <th class="tg-3pgb">Input Number</th>
> 
>     <th class="tg-3pgb">Envelopes</th>
> 
>     <th class="tg-3pgb">Index</th>
> 
>   </tr>
> 
> </thead>
> 
> <tbody>
> 
>   <tr>
> 
>     <td class="tg-ftz9" rowspan="2">0</td>
> 
>     <td class="tg-3guk"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-4ntm">i0</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-3guk"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-4ntm">i1</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-fvke">1</td>
> 
>     <td class="tg-ll5l"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-e6uf">i2</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-qk7o" rowspan="3">2</td>
> 
>     <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-edzw">i3</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-edzw">i4</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-edzw">i5</td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-fvke">3</td>
> 
>     <td class="tg-e6uf">none</td>
> 
>     <td class="tg-e6uf"></td>
> 
>   </tr>
> 
>   <tr>
> 
>     <td class="tg-mbap">4</td>
> 
>     <td class="tg-6jca"><img src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png" alt="Image" width="50" height="56"></td>
> 
>     <td class="tg-edzw">i6</td>
> 
>   </tr>
> 
> </tbody>
> 
> </table>
> 
> 

Exactly, that is the scheme!

**Comment by cypherpork** [comment]
`ord` peoples, comes without saying, it's yours to use.

**Comment by cypherpork** [comment]
@veryordinally, @raphjaph  – in the table above, 0 input has two envelopes: i0 and i1. 
The content of the envelope i0 is inscribed on the first satoshi of input 0.
On which satoshi is the i1 content inscribed?

**Comment by BillZong** [comment]
Currently, it is "cursed", and not bound to any satoshi.
The team describe it on tests instead of documentation.

Read the tests carefully, you should be able to get some corner cases including this one.

**Comment by raphjaph** [comment]
> @veryordinally - something like this?
> 
> Input Number	Envelopes	Index
> 0	<img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i0
> <img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i1
> 1	<img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i2
> 2	<img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i3
> <img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i4
> <img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i5
> 3	none	
> 4	<img alt="Image" width="50" height="56" src="https://raw.githubusercontent.com/cypherpork/CookBook/assetsetc/envelope50.png">	i6

Nice illustration! Will try to integrate into docs somewhere

**Comment by cypherpork** [comment]
> Currently, it is "cursed", and not bound to any satoshi.
> 
> The team describe it on tests instead of documentation.
> Read the tests carefully, you should be able to get some corner cases including this one.

*That's* where it is documented! 
Thank you, it will eliminate a lot of guesswork on my part :)

**Comment by casey** [comment]
This is now documented in [the handbook](https://docs.ordinals.com/inscriptions.html).
